R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,677 @@
+    "    }\n",
+    "    return(df)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Iterate over the dataset and call run the analyze function for each of the images:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datasetId <- REPLACE_WITH_DATASET_ID\n",
+    "\n",
+    "channelName <- 'CDK5RAP2-C'\n",
+    "\n",
+    "dataset <- loadObject(server, \"DatasetData\", datasetId)\n",
+    "\n",
+    "# Keep the channel name, image name, image id, area, perimeter, and diameter of the largest ROIs\n",
+    "result <- data.frame(Channel = c('remove'), ImageName = c('remove'), Image = c(0), ROIIndex = c(0), Area = c(0), Perimeter = c(0), Diameter = c(0), stringsAsFactors = FALSE)\n",
+    "\n",
+    "images <- getImages(dataset)\n",
+    "for (image in images) {\n",
+    "    result <- tryCatch({\n",
+    "        analyzeImage(image, channelName, result, saveROIs = TRUE)\n",
+    "    }, warning = function(war) {\n",
+    "        message(paste(\"WARNING:  \", image@dataobject$getId(),war))\n",
+    "        return(result)\n",
+    "    }, error = function(err) {\n",
+    "        message(paste(\"ERROR:  \", image@dataobject$getId() ,err))\n",
+    "        return(result)\n",
+    "    }, finally = {\n",
+    "    })\n",
+    "}\n",
+    "\n",
+    "result <- result[-1,]\n",
+    "rownames(result) <- c()\n",
+    "\n",
+    "# set the correct datatypes\n",
+    "result$Channel <- as.factor(result$Channel)\n",
+    "result$Area <- as.numeric(result$Area)\n",
+    "result$Perimeter <- as.numeric(result$Perimeter)\n",
+    "result$Diameter <- as.numeric(result$Diameter)\n",
+    "\n",
+    "result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Open some of the other images of the dataset in the full viewer to check the ROIs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Optional Exercise:** The measurements (Area, Perimeter and Diameter) are currently showing pixels. Find out the pixel size in nanometer (using OMERO.web) and apply it accordingly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# pxSizeInNM <- REPLACE_WITH_PIXEL_SIZE\n",
+    "# result$Area <- ?\n",
+    "# result$Perimeter <- ?\n",
+    "# result$Diameter <- ?\n",
+    "# result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Statistical analysis\n",
+    "\n",
+    "The image segmentation has already been run over the whole idr0021 project and the results attached to the project as table 'Summary from R'. \n",
+    "Instead of running the segmentation repeatedly we just load the table as dataframe 'centrioles':"
```

Comment:
``Instead of running the segmentation repeatedly we just load the table as dataframe 'centrioles' `` ->

``We are going to load this table from OMERO as dataframe "centrioles"``

Reply:
Maybe add: ``In this manner, we will quickly obtain a sizeable segmentation dataset to perform a statistical analysis on.``